### DISCUSSION TITLE
How to package without dcm4che source code

### DISCUSSION THREAD

**Discussion by Georgeqhh** [nroduit/Weasis#402] [Q&A] [answered]
In my project, I imported Weasis to print DICOM files and used DCM4CHE to perform other tasks related to DICOM. However, there were several issues caused by Weasis copying DCM4CHE's source code. Both libraries contain many identical classes, but their implementations may differ. How can I resolve this problem? Would it be possible to package Weasis without including the DCM4CHE source code?

**Comment by nroduit** [comment]
Please give examples (which classes are the same, what are the problems...) because without them it is not possible to give an answer.

**Reply by Georgeqhh** [reply]
classes: NativeImageReader，StreamSegment
problems: cannot find class or method(load the wrong class,the class has no this mothod,but another has),there are too many the same classes, so it is hard to change the class loaded sequence

**Comment by nroduit** [comment]
The classes you mentioned only exist in dcm4che project. I still don't see where the duplicate classes are.

**Reply by Georgeqhh** [reply]
![image](https://user-images.githubusercontent.com/16661018/236437855-deed7677-1494-48b5-936a-9670b0a18b52.png)
![image](https://user-images.githubusercontent.com/16661018/236438016-f33ac05a-defd-4805-b6e1-43a8af3e8e84.png)
weasis jar includes the dcm4che source code,weasis source code doesn't include,therefore,it is because I don't package weasis by rignt way?And What is the right package way?

**Reply by Georgeqhh** [reply, marked as answer]
It is because I package the dcm4che source code into weasis jar?
![image](https://user-images.githubusercontent.com/16661018/236441485-8ddad786-5c6e-46c3-b07b-6d55cd214b26.png)

**Comment by nroduit** [comment]
I can provide assistance within the scope of the Weasis project, but not to issues outside that scope with other objectives.
My priority is to support open-source developments that contribute to the common effort to improve Weasis.
